src/Agent.Listener/CommandSettings.cs

```diff
@@ -436,117 +453,230 @@ public string GetWork()
+        }
+
+        public bool GetDeploymentOrMachineGroup()
+        {
+            if (TestFlag(Configure.DeploymentGroup, Constants.Agent.CommandLine.Flags.DeploymentGroup) || 
+                (Configure != null && Configure.MachineGroup))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool Unattended()
+        {
+            if (TestFlag(Configure?.Unattended, Constants.Agent.CommandLine.Flags.Unattended) ||
+                    (Remove != null && Remove.Unattended))
+            {
+                return true;
+            }
+
+            return false;
         }
 
         //
-        // Private helpers.
+        // Command Checks
         //
-        private string GetArg(string name)
+        public bool IsRunCommand()
         {
-            string result;
-            if (!_parser.Args.TryGetValue(name, out result))
+            if (Run != null)
             {
-                result = GetEnvArg(name);
+                return true;
             }
 
-            return result;
+            return false;
+        }
+
+        public bool IsVersion()
+        {
+            if ((Configure != null && Configure.Version) ||
+                (Remove != null && Remove.Version) ||
+                (Run != null && Run.Version) ||
+                (Warmup != null && Warmup.Version))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsHelp()
+        {
+            if ((Configure != null && Configure.Help) ||
+                (Remove != null && Remove.Help) ||
+                (Run != null && Run.Help) ||
+                (Warmup != null && Warmup.Help))
+            {
+                return true;
+            }
+
+            return false;
         }
 
-        private void RemoveArg(string name)
+        public bool IsCommit()
         {
-            if (_parser.Args.ContainsKey(name))
+            if (Run != null && Run.Commit)
             {
-                _parser.Args.Remove(name);
+                return true;
             }
 
-            if (_envArgs.ContainsKey(name))
+            return false;
+        }
+
+        public bool IsRunDiagnostics()
+        {
+            if (Run != null && Run.Diagnostics)
             {
-                _envArgs.Remove(name);
+                return true;
             }
+
+            return false;
+        }
+
+        public bool IsConfigureCommand()
+        {
+            if (Configure != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsRemoveCommand()
+        {
+            if (Remove != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsWarmupCommand()
+        {
+            if (Remove != null)
```

Comment:
Warmup?

Reply:
Whoopie, I'll fix this and add a test.  Great catch!

Reply:
maybe we need to make sure we have a unit test cover each of the Is*Command methods